rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -54,16 +54,16 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.ReentrantLock;
 
 public class LongRunningAction extends BasicAction {
     private static final String LRA_TYPE = "/StateManager/BasicAction/LongRunningAction";
-    private final ScheduledExecutorService scheduler;
+    private static final ScheduledExecutorService scheduler = new ScheduledThreadPoolExecutor(10);
```

Comment:
Although 10 seems low, the scheduler only runs for LRAs that time out at the same time. In a typical user application I'd i) expect most LRAs to be explicitly finished, and (ii) it's unlikely that lot's of LRAs will time out at similar deadlines.

Reply:
The number of available `hardware threads` is restricted to the number of cores times the hyperthreading.

BTW the size of the pool relates to software threads and there can be hundreds of those (they are managed by the kernel and they also increase parallelism - eg if the thread is executing I/O bound code then there will be a context switch so that other code can be executed.